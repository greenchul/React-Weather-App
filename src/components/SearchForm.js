import React from "react";
import propTypes from "prop-types";

const SearchForm = ({ handleSearchSelect, setSearchedLocation }) => {
  return (
    <div className="search-form">
      <input
        className="search-form__input"
        placeholder="City"
        onChange={(e) => {
          setSearchedLocation(e.target.value);
        }}
      />
      <button
        type="button"
        className="search-form__button"
        onClick={() => {
          handleSearchSelect();
        }}
      >
        Search
      </button>
    </div>
  );
};

SearchForm.propTypes = {
  handleSearchSelect: propTypes.func.isRequired,
  setSearchedLocation: propTypes.func.isRequired,
};

export default SearchForm;
