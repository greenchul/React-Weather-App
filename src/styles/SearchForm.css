.search-form__input {
  font-size: 15px;
}

@media screen and (min-width: 768px) {
  .search-form__input {
    font-size: 20px;
  }
}

@media screen and (min-width: 1024px) {
}
