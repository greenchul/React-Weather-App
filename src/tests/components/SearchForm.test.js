import React from "react";
import { render, screen } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm component", () => {
  const validProps = {
    handleSearchSelect: () => {},
    setSearchedLocation: () => {},
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        handleSearchSelect={validProps.handleSearchSelect}
        setSearchedLocation={validProps.setSearchedLocation}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("button renders correctly", () => {
    render(
      <SearchForm
        handleSearchSelect={validProps.handleSearchSelect}
        setSearchedLocation={validProps.setSearchedLocation}
      />
    );
    const buttonElement = screen.getByText("Search");

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveClass("search-form__button");
  });

  it("input renders correctly", () => {
    render(
      <SearchForm
        handleSearchSelect={validProps.handleSearchSelect}
        setSearchedLocation={validProps.setSearchedLocation}
      />
    );
    const inputElement = screen.getByPlaceholderText("City");

    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveClass("search-form__input");
  });
});
