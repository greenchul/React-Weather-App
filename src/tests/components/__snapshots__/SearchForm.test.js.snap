// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      class="search-form__input"
      placeholder="City"
    />
    <button
      class="search-form__button"
      type="button"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
